services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedTenantService.java

```diff
@@ -0,0 +1,630 @@
+     * <p>
+     * Tenant modification have to be enabled in {@link MongoDbBasedTenantsConfigProperties} or exits with {@code HttpURLConnection.HTTP_FORBIDDEN}.
+     * </p>
+     * <p>
+     * Provided version {@code resourceVersion} have to be empty or equal to the version of present tenant.
+     * A provided <em>certificate authority</em> must not be set in other tenants.
+     * </p>
+     *
+     * @param tenantId        The identifier of the tenant.
+     * @param tenantObj       The updated configuration information for the tenant (may be {@code null}).
+     * @param resourceVersion The identifier of the resource version to update.
+     * @param span            The active OpenTracing span for this operation. It is not to be closed in this method!
+     *                        An implementation should log (error) events on this span and it may set tags and use this span as the
+     *                        parent for any spans created in this method.
+     * @return Result of update operation: On success {@code HttpURLConnection.HTTP_NO_CONTENT}, http error codes else. {@code HttpURLConnection.HTTP_FORBIDDEN} if modification is disabled.
+     * @throws NullPointerException if {@code tenantId} or {@code tenantObj} are {@code null}.
+     */
+    @Override
+    public Future<OperationResult<Void>> updateTenant(final String tenantId, final Tenant tenantObj, final Optional<String> resourceVersion, final Span span) {
+        Objects.requireNonNull(tenantId);
+        Objects.requireNonNull(tenantObj);
+
+        return executeIfModificationEnabled(
+                () -> processUpdateTenant(tenantId, tenantObj, resourceVersion, span),
+                span
+        );
+    }
+
+    /**
+     * Updates data of a present tenant.
+     * <p>
+     * Provided version {@code resourceVersion} have to be empty or equal to the version of present tenant.
+     * A provided <em>certificate authority</em> must not be set in other tenants.
+     * </p>
+     *
+     * @param tenantId        The identifier of the tenant.
+     * @param tenantObj       The updated configuration information for the tenant (may be {@code null}).
+     * @param resourceVersion The identifier of the resource version to update.
+     * @param span            The active OpenTracing span for this operation. It is not to be closed in this method!
+     *                        An implementation should log (error) events on this span and it may set tags and use this span as the
+     *                        parent for any spans created in this method.
+     * @return Result of update operation: On success {@code HttpURLConnection.HTTP_NO_CONTENT}, http error codes else.
+     * @throws NullPointerException if {@code tenantId} or {@code tenantObj} are {@code null}.
+     */
+    private Future<OperationResult<Void>> processUpdateTenant(final String tenantId, final Tenant tenantObj, final Optional<String> resourceVersion, final Span span) {
```

Comment:
Please follow the hono `javadoc` style and you could use `TenantManagementService` as a reference. Please check the _javadoc_ of other methods too.